UnitTests/Freemium/DBP/FreemiumDBPPromotionViewCoordinatingTests.swift

```diff
@@ -0,0 +1,271 @@
+//
+//  FreemiumDBPPromotionViewCoordinatingTests.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+import Freemium
+@testable import DuckDuckGo_Privacy_Browser
+import Combine
+
+final class FreemiumDBPPromotionViewCoordinatingTests: XCTestCase {
+
+    private var sut: FreemiumDBPPromotionViewCoordinator!
+    private var mockUserStateManager: MockFreemiumDBPUserStateManager!
+    private var mockFeature: MockFreemiumDBPFeature!
+    private var mockPresenter: MockFreemiumDBPPresenter!
+    private let notificationCenter: NotificationCenter = .default
+    private var cancellables: Set<AnyCancellable> = []
+
+    @MainActor
```

Comment:
Do these annotations have effect if none of the test functions are marked as `async`?

Reply:
Yes. The functions under test are not `async`, but the overall type `FreemiumDBPPromotionViewCoordinator` is annotated as `@MainActor` (via it’s protocol `FreemiumDBPPromotionViewCoordinating`), and so the test methods need to be `MainActor`